src/card_operations/check_pairing.c

```diff
@@ -0,0 +1,147 @@
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * TODO: add description when this api gets concrete definition
+ *
+ * @param connection_data
+ */
+static void init_tap_card_data(NFC_connection_data *connection_data);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+static void init_tap_card_data(NFC_connection_data *connection_data) {
+  memset(connection_data, 0, sizeof(NFC_connection_data));
+  connection_data->retries = 5;
+  connection_data->desktop_control = false;
+  connection_data->acceptable_cards = ACCEPTABLE_CARDS_ALL;
+  memset(connection_data->family_id, 0xff, sizeof(connection_data->family_id));
+  memset(connection_data->card_key_id, 0, sizeof(connection_data->card_key_id));
+}
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+card_error_type_e card_check_pairing(check_pairing_result_t *result) {
+  LOG_SWV("%s (%d): %p\n", __func__, __LINE__, result);
+  card_error_type_e status = CARD_OPERATION_DEFAULT_INVALID;
+  if (NULL == result) {
+    return status;
+  }
+
+  memset(result, 0, sizeof(check_pairing_result_t));
+  result->card_number = 0;
+  card_operation_data_t operation_data = {tap_card_data, NULL, 0};
+  init_tap_card_data(&operation_data.nfc_data);
+  status = card_initialize_applet(&operation_data);
+
+  if (CARD_OPERATION_SUCCESS == status) {
+    result->card_number =
+        decode_card_number(operation_data.nfc_data.tapped_card);
+    if (-1 == is_paired(operation_data.nfc_data.card_key_id)) {
+      // keystore index is -1; the keystore does not have key-id
+      result->is_paired = false;
+    } else {
+      result->is_paired = true;
+    }
+  } else {
+    if (NULL != operation_data.error_message) {
+      delay_scr_init(operation_data.error_message, DELAY_TIME);
+    }
+    LOG_SWV("%s (%d):", __func__, __LINE__);
+    LOG_CRITICAL("COCP %04X\n", operation_data.nfc_data.status);
+  }
+  buzzer_start(100);
+  return status;
```

Comment:
On return, please call nfc_deselect_card to turn off the NFC session.

Reply:
added